operator/standalone/delete_it_test.go

```diff
@@ -0,0 +1,326 @@
+			},
+			givenNamespace: "test-namespace",
+			givenSecret:    "",
+			expectedError:  "resource name may not be empty",
+		},
+	}
+	for name, tc := range tests {
+		ts.Run(name, func() {
+			d := &DeleteStandalonePipeline{
+				client: ts.Client,
+				instance: newBuilderInstance("instance", tc.givenNamespace).
+					setConnectionSecret(tc.givenSecret).
+					get(),
+				helmReleaseDeleted: false,
+			}
+			tc.prepare(tc.givenSecret, tc.givenNamespace)
+			err := d.deleteConnectionSecret(ts.Context)
+
+			// Assert
+			if tc.expectedError != "" {
+				ts.Assert().EqualError(err, tc.expectedError)
+				return
+			}
+			ts.Assert().NoError(err)
+			resultSecret := &corev1.Secret{}
+			err = ts.Client.Get(
+				ts.Context,
+				types.NamespacedName{Name: tc.givenSecret, Namespace: tc.givenNamespace},
+				resultSecret,
+			)
+			ts.AssertResourceNotExists(resultSecret.GetDeletionTimestamp(), err)
+		})
+	}
+}
+
+func (ts *DeleteStandalonePipelineSuite) Test_RemoveFinalizer() {
+	tests := map[string]struct {
+		prepare        func(instance *v1alpha1.PostgresqlStandalone)
+		givenInstance  string
+		givenNamespace string
+	}{
+		"GivenAnInstance_WhenDeletingFinalizer_ThenExpectInstanceWithNoFinalizer": {
+			prepare: func(instance *v1alpha1.PostgresqlStandalone) {
+				ts.EnsureNS("remove-finalizer")
+				ts.EnsureResources(instance)
+			},
+
+			givenInstance:  "instance",
+			givenNamespace: "remove-finalizer",
+		},
+	}
+	for name, tc := range tests {
+		ts.Run(name, func() {
+			instance := newBuilderInstance(tc.givenInstance, tc.givenNamespace).setFinalizers(finalizer).get()
+			d := &DeleteStandalonePipeline{
+				client:             ts.Client,
+				instance:           instance,
+				helmReleaseDeleted: false,
+			}
+			tc.prepare(instance)
+			err := d.removeFinalizer(ts.Context)
+			// Assert
+			ts.Require().NoError(err)
+			releaseResult := &helmv1beta1.Release{}
+			err = ts.Client.Get(
+				ts.Context,
+				types.NamespacedName{Name: tc.givenInstance, Namespace: tc.givenNamespace},
+				releaseResult,
+			)
+			ts.AssertResourceNotExists(releaseResult.GetDeletionTimestamp(), err)
+			ts.Assert().Empty(releaseResult.Finalizers)
+		})
+	}
+}
+
+func newPostgresqlHelmRelease(name string) *helmv1beta1.Release {
+	return &helmv1beta1.Release{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+}
+
+func newPostgresqlInstance(name, namespace string) v1alpha1.PostgresqlStandalone {
+	return v1alpha1.PostgresqlStandalone{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PostgresqlStandaloneSpec{
+			Parameters: v1alpha1.PostgresqlStandaloneParameters{
+				MajorVersion:    v1alpha1.PostgresqlVersion14,
+				EnableSuperUser: true,
+				Resources: v1alpha1.Resources{
+					ComputeResources: v1alpha1.ComputeResources{MemoryLimit: parseResource("2Gi")},
+					StorageResources: v1alpha1.StorageResources{StorageCapacity: parseResource("1Gi")},
+				},
+			},
+		},
+	}
+}
```

Comment:
We already have this function almost identical in `create_test.go`. Can we consolidate that?

Reply:
I think I considered that function and discarded it back then, let me check